pkg/splunk/enterprise/configuration.go

```diff
@@ -875,17 +877,120 @@ func AreRemoteVolumeKeysChanged(client splcommon.ControllerClient, cr splcommon.
 	return false
 }
 
+// ApplyManualAppUpdateConfigMap applies the manual app update config map
+func ApplyManualAppUpdateConfigMap(client splcommon.ControllerClient, cr splcommon.MetaObject, crKindMap map[string]string) (*corev1.ConfigMap, error) {
+
+	scopedLog := log.WithName("ApplyManualAppUpdateConfigMap").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+
+	configMapName := GetSplunkManualAppUpdateConfigMapName()
+
+	scopedLog.Info("Creating/Updating manual app update configMap")
+
+	configMap := splctrl.PrepareConfigMap(configMapName, cr.GetNamespace(), crKindMap)
+
+	// Owner ref doesn't exist, update configMap with owner references
+	if getManualUpdateRefCount(crKindMap[cr.GetObjectKind().GroupVersionKind().Kind]) == 1 {
+		configMap.SetOwnerReferences(append(configMap.GetOwnerReferences(), splcommon.AsOwner(cr, false)))
+	}
+
+	_, err := splctrl.ApplyConfigMap(client, configMap)
+	if err != nil {
+		scopedLog.Error(err, "Unable to apply the configMap", "name", configMap)
+		return nil, err
+	}
+
+	return configMap, nil
+}
+
+// getManualUpdateStatus extracts the status field from the configMap data
+func getManualUpdateStatus(data string) string {
+	statusRegex := ".*status: (?P<status>.*).*"
+	pattern := regexp.MustCompile(statusRegex)
+	if len(pattern.FindStringSubmatch(data)) > 0 {
+		return pattern.FindStringSubmatch(data)[1]
+	}
+	return ""
+}
+
+// getManualUpdateRefCount extracts the refCount field from the configMap data
+func getManualUpdateRefCount(data string) int {
+	var refCount int
+	refCountRegex := ".*refCount: (?P<refCount>.*).*"
+	pattern := regexp.MustCompile(refCountRegex)
+	if len(pattern.FindStringSubmatch(data)) > 0 {
+		refCount, _ = strconv.Atoi(pattern.FindStringSubmatch(data)[1])
+	}
+	return refCount
+}
+
+// createOrUpdateAppUpdateConfigMap creates or updates the manual app update configMap
+func createOrUpdateAppUpdateConfigMap(client splcommon.ControllerClient, cr splcommon.MetaObject) (*corev1.ConfigMap, error) {
```

Comment:
Are there any issue with how this configmap is handled if the operator pod restarts?

Reply:
I haven't noticed any issues while starting/stopping/starting debugging the operator, so hopefully no issues in real-world scenarios too :)